crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -790,12 +782,44 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
             }
         }
 
-        // check unconnected block buffer for the childs of new added blocks,
-        for added_block in last_canonical_hashes.into_iter() {
+        self.connect_buffered_blocks_to_hashes(last_canonical_hashes)?;
+
+        Ok(())
+    }
+
+    /// Reads the last `N` canonical hashes from the database and updates the block indices of the
+    /// tree by attempting to connect the buffered blocks to canonical hashes.
+    ///
+    /// `N` is the `max_reorg_depth` plus the number of block hashes needed to satisfy the
+    /// `BLOCKHASH` opcode in the EVM.
```

Comment:
I don't understand why we need this?

why do we need to store additional canonical hashes in the tree?

Reply:
from my other PR, if I'm not mistaken: https://github.com/paradigmxyz/reth/blob/9fde694bbd026beeb16ee6739ca0190eeaa47635/crates/blockchain-tree/src/config.rs#L78-L82